lessons/specifics/iex_helpers.md

```diff
@@ -0,0 +1,250 @@
+---
+layout: page
+title: IEx Helpers
+category: specifics
+order: 6
+lang: en
+---
+
+{% include toc.html %}
+
+## Overview
+As you begin to work in Elixir, IEx is your best friend.
+It is a REPL, but it has many advanced features that can make life easier when exploring new code or developing your own work as you go.
+There are a slew of built-in helpers that we will go over in this lesson.
+
+### Autocomplete
+When working in the shell, you often might find yourself using a new module that you are unfamiliar with.
+To understand some of what is available to you, the autocomplete functionality is wonderful.
+Simply type a module name and then `.` and press `Tab`:
+
+```elixir
+iex> Map. # press Tab
+delete/2             drop/2               equal?/2
+fetch!/2             fetch/2              from_struct/1
+get/2                get/3                get_and_update!/3
+get_and_update/3     get_lazy/3           has_key?/2
+keys/1               merge/2              merge/3
+new/0                new/1                new/2
+pop/2                pop/3                pop_lazy/3
+put/3                put_new/3            put_new_lazy/3
+split/2              take/2               to_list/1
+update!/3            update/4             values/1
+```
+
+And now we know the functions we have and their arity!
+
+### `.iex.exs`
+Every time IEx starts, it will look for a `.iex.exs` file.
```

Comment:
I'm not sure about this example, what do you think about cleaning this up a bit:

"Every time IEx starts it will look for a `.iex.exs` configuration file.  If it's not present in the current directory, then the user's home directory (`~/.iex.exs`) will be used as the fallback.  

Configuration options and code defined within this file will be available to us when the IEx shell starts up.  For instance if we want some helper functions available to us in IEx, we can open up `.iex.exs` and make some changes.

Let's start by adding a module with a few helper methods:

```elixir
defmodule IExHelpers do
  def whats_this?(term) when is_nil(term), do: "Type: Nil"
  def whats_this?(term) when is_binary(term), do: "Type: Binary"
  def whats_this?(term) when is_boolean(term), do: "Type: Boolean"
  def whats_this?(term) when is_atom(term), do: "Type: Atom"
  def whats_this?(_term), do: "Type: Unknown"
end
```

Now when we run IEx we'll have the `IExHelpers` module available to us from the start.  Open up IEx and let's try out our new helpers:

```shell
$ iex
{{ site.erlang.OTP }} [{{ site.erlang.erts }}] [source] [64-bit] [smp:8:8] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]

Interactive Elixir ({{ site.elixir.version }}) - press Ctrl+C to exit (type h() ENTER for help)
iex> IExHelpers.whats_this?("a string")
"Type: Binary"
iex> IExHelpers.whats_this?(%{})
"Type: Unknown"
iex> IExHelpers.whats_this?(:test)
"Type: Atom"
```

As we can see we don't need to do anything special to require or import our helpers, IEx handles that for us."